src/sia/sentry.py

```diff
@@ -0,0 +1,58 @@
+"""Sentry integration helpers."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import Any
+
+from safir.sentry import before_send_handler
+
+from .config import config
+
+__all__ = ["enable_sentry", "make_traces_sampler"]
+
+
+def enable_sentry() -> None:
+    """Enable Sentry telemetry."""
+    try:
+        import sentry_sdk
+    except ImportError:
+        return
+
+    sentry_sdk.init(
+        dsn=config.sentry_dsn,
+        environment=config.environment_name,
+        before_send=before_send_handler,
+        traces_sampler=make_traces_sampler(config.sentry_traces_sample_rate),
+    )
+
+
+def make_traces_sampler(
+    original_rate: float,
+) -> Callable[[dict[str, Any]], float]:
+    """
+    Create a traces sampler that samples all requests except for health checks.
+
+    Parameters
+    ----------
+    original_rate
+        The original sample rate.
+
+    Returns
+    -------
+    Callable[[dict[str, Any]], float]
+        The traces sampler.
+    """
+
+    def traces_sampler(context: dict[str, Any]) -> float:
+        asgi_scope = context.get("asgi_scope")
+        if not asgi_scope:
+            return original_rate
+        if (
+            asgi_scope.get("path", "").startswith("/health")
+            or asgi_scope.get("path") == "/"
+        ):
+            return 0
+        return original_rate
+
+    return traces_sampler
```

Comment:
Nice!

* Sentry will drop (and not count against our quota) any transactions coming from [these endpoints](https://docs.sentry.io/concepts/data-management/filtering/#transactions-coming-from-health-check), so if you're gonna change the kube probe URL to `/healthcheck`, then we shouldn't see the telemetry. This does still do the tracing in the app though, whereas this sampler function will prevent the app from starting the transaction entirely, which maybe is nice for performance reasons (though hopefully tracing the healthcheck doesn't make too much of an impact...)
* If you do want to do it this way, it could be nice to lift into Safir at some point, a function to take a list of strings and/or regexes and the original sample rate and ignores those routes.

Reply:
Ah ok thanks for that I wasn't aware that Sentry automatically dropped those paths.
I guess I could then just probably remove all of that, though I'm leaning towards leaving it in because I think I will end up adding some exceptions for certain paths, depending on their traffic which i need to investigate (/availability & /capabilities)

I've slightly modified the sampler method based on your suggestions, and will test it now with a couple of debug endpoints, let me know if you think it looks ok, or if you have any suggestions for changes to it.

Reply:
Looks good, nice!